### ISSUE TITLE
Unhandled exception. FlubuCore.Scripting.ScriptLoaderExcetpion

### ISSUE THREAD

**Issue by sreejith-ms** [dotnetcore/FlubuCore#347]
**Describe the bug**
Getting error when running command `flubu` on root directory with a `.flubu` file

**To Reproduce**
1. Using this version of the library 'Flubu v.6.1.1.0'
2. Run this code
```
public class BuildScript : DefaultBuildScript
{
    [SolutionFileName]
    public string SolutionFileName { get; set; } = "ConsoleApp3.sln";

    protected override void ConfigureTargets(ITaskContext context)
    {
        context.LogInfo("started");
        var compile = context.CreateTarget("compile")
            .SetDescription("Compiles the solution.")
            .AddCoreTask(x => x.Build());

        var clean = context.CreateTarget("Clean")
            .SetDescription("Cleans the output of all projects in the solution.")
            .DependsOn(compile)
            .AddCoreTask(x => x.Clean());

        var restore = context.CreateTarget("Restore")
            .SetDescription("Restores the dependencies and tools of all projects in the solution.")
            .DependsOn(clean)
            .AddCoreTask(x => x.Restore());

        context.CreateTarget("Default")
            .SetDescription("Runs all targets.")
            .SetAsDefault()
            .DependsOn(restore);
    }
}
```
.flubu file `BuildScript/BuildScript.csproj`
3. With these arguments `compile`
```
Flubu v.6.1.1.0
using the build script file path from .flubu file. 'BuildScript/BuildScript.csproj'.
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '/'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1002: ; expected
ScriptError:CS1002: ; expected
ScriptError:CS1525: Invalid expression term '>'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '/'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1002: ; expected
ScriptError:CS1002: ; expected
ScriptError:CS1525: Invalid expression term '>'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '/'
ScriptError:CS1525: Invalid expression term '<'
ScriptError:CS1525: Invalid expression term '/'
ScriptError:CS1733: Expected expression
ScriptError:CS1002: ; expected
ScriptError:CS8805: Program using top-level statements must be an executable.
ScriptError:CS0103: The name 'Project' does not exist in the current context
ScriptError:CS0103: The name 'Sdk' does not exist in the current context
ScriptError:CS0103: The name 'PropertyGroup' does not exist in the current context
ScriptError:CS0103: The name 'TargetFramework' does not exist in the current context
ScriptError:CS0103: The name 'netcoreapp3' does not exist in the current context
ScriptError:CS0103: The name 'TargetFramework' does not exist in the current context
ScriptError:CS0103: The name 'PropertyGroup' does not exist in the current context
ScriptError:CS0103: The name 'ItemGroup' does not exist in the current context
ScriptError:CS0103: The name 'PackageReference' does not exist in the current context
ScriptError:CS0103: The name 'Include' does not exist in the current context
ScriptError:CS0103: The name 'Version' does not exist in the current context
ScriptError:CS0103: The name 'ItemGroup' does not exist in the current context
ScriptError:CS0103: The name 'ItemGroup' does not exist in the current context
ScriptError:CS0103: The name 'DotNetCliToolReference' does not exist in the current context
ScriptError:CS0103: The name 'Include' does not exist in the current context
ScriptError:CS0103: The name 'Version' does not exist in the current context
ScriptError:CS0103: The name 'ItemGroup' does not exist in the current context
ScriptError:CS0103: The name 'Project' does not exist in the current context
Unhandled exception. FlubuCore.Scripting.ScriptLoaderExcetpion: Csharp source code file: BuildScript/BuildScript.csproj has some compilation errors! If your script doesn't have compilation errors in VS or VSCode script probably doesn't include .cs file. To resolve this issue you should see build script fundamentals section 'Adding other .cs files to script' for more details: https://flubucore.dotnetcore.xyz/referencing-external-assemblies/#adding-other-cs-files-to-script
   at FlubuCore.Scripting.ScriptLoader.CompileBuildScript(String buildScriptAssemblyPath, String buildScriptFilePath, IEnumerable`1 references, String code) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 250
   at FlubuCore.Scripting.ScriptLoader.FindAndCreateBuildScriptInstanceAsync(CommandArguments args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 124
   at FlubuCore.Scripting.ScriptProvider.GetBuildScriptAsync(CommandArguments commandArguments, Boolean forceReload) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptProvider.cs:line 26
   at DotNet.Cli.Flubu.Program.FlubuStartup(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 84
   at DotNet.Cli.Flubu.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 36
   at FlubuCore.Tool.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore.Tool\Program.cs:line 9
   at FlubuCore.Tool.Program.<Main>(String[] args)
```

**Comment by sreejith-ms** [comment]
Another point is the path in the stacktrace does not exist.
`at FlubuCore.Tool.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore.Tool\Program.cs:line 9`
Jenkins is not installed on my machine ;)

**Comment by mzorec** [comment]
I think that the .flubu file in your scenario has wrong data. How did you create .flubu file?

.flubu file should contain at line 0 path to build script. For example: BuildScript/BuildScript.cs
.flubu file should contain at line 1 path to project file. For example BuildScript/BuildScript.csproj

From your log it looks like that it is 'BuildScript/BuildScript.csproj' is at line 0 in .flubu file and because of that it tries to compile csproj file instead of .cs file.

.flubu file can contain only path to build script file or path to project file but it must be at correct lines in .flubu file.

.flubu file can also be empty. Then it will look for both files at default locations. Actually it doesn't need to exist at all but it has some benefits which I would not go into details at the moment.

**Comment by mzorec** [comment]
You can see sample of .flubu file with only location to csproj file here https://github.com/dotnetcore/CAP/blob/master/.flubu

**Comment by sreejith-ms** [comment]
Thanks, @mzorec 
I have created the .flubu file from the same CAP repo, but when I executed flubu command I'm got the below error so I removed the new line from the file.
```
Script file path was not explicitly specified, searching the default locations.
EXECUTION FAILED:
The build script file was not found.  Please specify location of the build script in the argument '-s={PathToBuildScript}' or run 'flubu setup' to store the location in a file or use some of the default paths for script file:
```

Can you show the documentation of .flubu file?

Now when I changed the .flubu file to 
```
BuildScript/BuildScript.cs
BuildScript/BuildScript.csproj
```
the sample script is running. 

But on my actual project, it is still failing.

```
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations
ScriptError:CS0146: Circular base type dependency involving 'BuildScript' and 'BuildScript'
ScriptError:CS0246: The type or namespace name 'Versionize' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Repository' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Commit' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Commit' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Regex' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Repository' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Tag' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Repository' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Tag' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Repository' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Tag' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Commit' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'DirectoryInfo' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'DirectoryInfo' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonPropertyNameAttribute' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonPropertyName' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonPropertyNameAttribute' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonPropertyName' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonPropertyNameAttribute' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonPropertyName' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonIgnoreAttribute' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'JsonIgnore' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0246: The type or namespace name 'Regex' could not be found (are you missing a using directive or an assembly reference?)
ScriptError:CS0103: The name 'RegexOptions' does not exist in the current context
ScriptError:CS0103: The name 'RegexOptions' does not exist in the current context
ScriptError:CS0103: The name 'RegexOptions' does not exist in the current context
```

I have a reference to `LibGit2Sharp` and some logic to find out the version.

**Comment by sreejith-ms** [comment]
In this [doc](https://flubucore.dotnetcore.xyz/referencing-external-assemblies/) it is mentioned that nuget packages automatically from build script csproj. 

> Csproj must be at on of the location specified here If not assembly and nuget references will not be loaded automatically when executing script.

The location mentioned in the documentation does not exist.

https://github.com/dotnetcore/FlubuCore/blob/master/FlubuCore/Scripting/Analysis/ProjectFileAnalyzer.cs


This is my csproj

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="FlubuCore" Version="5.1.8" />
    <PackageReference Include="LibGit2Sharp" Version="0.26.2" />
  </ItemGroup>

  <ItemGroup>
    <DotNetCliToolReference Include="dotnet-flubu" Version="5.1.8" />
  </ItemGroup>
</Project>
```

**Comment by mzorec** [comment]
Can you send me the actual buildscript file? If you don't want to publish it here you can send it on my email mzorec@gmail.com

**Comment by mzorec** [comment]
https://github.com/dotnetcore/FlubuCore/blob/develop/src/FlubuCore/Scripting/Analysis/ProjectFileAnalyzer.cs Correct file location is here. I will fix the documentation.

But if you have location of csproj file defined in .flubu file it doesn't need to be at the default locations defined in ProjectFileAnalyzer.cs

I think that the root of the problem might be this error 

ScriptError:CS1529: A using clause must precede all other elements defined in the namespace except extern alias declarations

That's why I would like to see your buildscript.

**Comment by sreejith-ms** [comment]
@mzorec Please take a look at here https://github.com/sreejith-ms/flubucore-versionize-sample

I have fixed few issues. One issue is the placement of `using` statements, previously I have placed the using statements under the namespace (following an .editorconfig rule IDE0065). 
```
namespace BuildScript
{
    using System;
    using System.Collections.Generic;
}
```
When I moved this to above the namespace some script errors disappeared.
Another issue is the usage of an `interface` produced this exception,
```
ScriptError:CS0101: The namespace '<global namespace>' already contains a definition for 'BuildScript'
Unhandled exception. FlubuCore.Scripting.ScriptLoaderExcetpion: Csharp source code file: BuildScript/BuildScript.cs has some compilation errors!
   at FlubuCore.Scripting.ScriptLoader.CompileBuildScript(String buildScriptAssemblyPath, String buildScriptFilePath, IEnumerable`1 references, String code) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 250
   at FlubuCore.Scripting.ScriptLoader.FindAndCreateBuildScriptInstanceAsync(CommandArguments args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 124
   at FlubuCore.Scripting.ScriptProvider.GetBuildScriptAsync(CommandArguments commandArguments, Boolean forceReload) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptProvider.cs:line 26
   at DotNet.Cli.Flubu.Program.FlubuStartup(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 84
   at DotNet.Cli.Flubu.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 36
   at FlubuCore.Tool.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore.Tool\Program.cs:line 9
   at FlubuCore.Tool.Program.<Main>(String[] args)
``` 
so I removed the interface.

Now I'm getting an exception from LibGit2Sharp
```
ERROR: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception.
 ---> System.DllNotFoundException: Unable to load DLL 'git2-106a5f2' or one of its dependencies: The specified module could not be found. (0x8007007E)
```

**Comment by mzorec** [comment]
Flubu is having problem loading LibGit2Sharp.NativeBinaries package which contains compiled versions of the libgit2 native library. Well it has problem loading any native library not only from libgit2. Not sure how fast I will be able to fix this issue.

If you need a fast workaround one would be to put everything related with LibGit2Sharp in a seperate console application (Let's call it MyConsoleApp.exe) and then run that console application from flubu script

```c#
  context.CreateTarget("Sample")
                .AddTask(x => x.RunProgramTask("Console/MyConsoleApp.exe")
                .WithArguments("Put any arguments to your console application here if needed."));
```

**Comment by sreejith-ms** [comment]
@mzorec I'm on a similar path, instead of running it in a console app I'm trying to run Flubu itself from a console app like described [here](https://flubucore.dotnetcore.xyz/execute-script-in-console-app-or-pack-as-global-tool/).

**Comment by sreejith-ms** [comment]
@mzorec I have updated the script to test dynamic target creation but it's not working. 
Did you know whether it is supported or how to handle this usecase?

```
var projectFiles = context.GetFiles(RootDirectory, "**/*.csproj");
var targets = new List<ITarget>
{
    compile,
    clean,
    restore
};

foreach (var project in projectFiles)
{
    var build = context.CreateTarget($"Build{project.FileName}")
        .SetDescription($"Builds {project.FileName} in the solution.")
        .DependsOn(restore)
        .AddCoreTask(x => x.Build()
            .Project(project.ToFullPath())
            .InformationalVersion("1.0.0"));

    var pack = context.CreateTarget($"Pack{project.FileName}")
        .SetDescription($"Creates nuget package for {project.FileName}")
        .DependsOn(build)
        .AddCoreTask(x => x.Pack()
            .NoBuild()
            .IncludeSymbols()
            .VersionSuffix(VersionSuffix)
            .OutputDirectory(ArtifactsDir));

    targets.Add(build);
    targets.Add(pack);
}

context.CreateTarget("Default")
    .SetDescription("Runs all targets.")
    .SetAsDefault()
    .DependsOn(targets.ToArray());
```
It breaks on NuGet packaging on 1st iteration, but 1st iteration is a success.

```
        Successfully created package 'D:\flubucore-versionize-sample\artifacts\BuildScript.1.0.0.nupkg'.
        Successfully created package 'D:\flubucore-versionize-sample\artifacts\BuildScript.1.0.0.symbols.nupkg'.
      External program C:\Program Files\dotnet\dotnet.exe failed with 1.
      PackBuildScript.csproj failed (took 2 seconds)
   Default failed (took 16 seconds)
ERROR: External program C:\Program Files\dotnet\dotnet.exe failed with 1.


Target                   Duration  Status
-----------------------------------------------
Compile                  00:00:04  Succeeded
Clean                    00:00:02  Succeeded
Restore                  00:00:02  Succeeded
BuildBuildScript.csproj  00:00:04  Succeeded
PackBuildScript.csproj   00:00:02  Failed
Default                  00:00:16  Failed

BUILD FAILED
```

**Comment by mzorec** [comment]
Yes that will work. I have tested it in my test console app and it worked. Problem with this solution might be that if you will run your build on build server you will have to compile console app locally and commit it. or first run dotnet build to compile the console app and then run the build with console app.

**Comment by mzorec** [comment]
Didn't notice your question before.

> Can you show the documentation of .flubu file?

I've noticed  .flubu file is not documented good enough. I will add documentation for it to getting started guide

.flubu file can contain path to buildscript and csproj file as mentioned in one of my comments above. It is also used to determinate the root directory as described here

https://flubucore.dotnetcore.xyz/execute-script-in-console-app-or-pack-as-global-tool/

FlubuCore command line tool has an option to help you create .flubu file, simply type flubu setup and interactive mode will guide you through.

**Comment by sreejith-ms** [comment]
Thanks, @mzorec :clap:

Two more questions from my side :blush:

Why usage of an `interface` and the placement of `using` statements throw script errors?

**Comment by mzorec** [comment]
@sreejith-ms Flubu removes namespace from .cs before script compilation. Script is compiled with roslyn and namespaces are not allowed in script files. I will fix the issue if using statements are placed after a namespace. 

Can you be more specific what you mean by usage of an interface throw script error? Usage of interfaces as I know should work.

**Comment by mzorec** [comment]
oh and one more suggestion. When creating new scripts I recomend  using flubu templates with command  `flubu new {TemplateName} `. You can found more details here https://flubucore.dotnetcore.xyz/flubu-templates/

It is also possible to generate continous integration configuration files (jenkins, azure pipelines, github actions, travis..) so you don't have to create them manually. More details here: https://flubucore.dotnetcore.xyz/CI-Generation/

**Comment by sreejith-ms** [comment]
@mzorec I have updated the [flubu sample repo](https://github.com/sreejith-ms/flubucore-versionize-sample/blob/master/BuildScript/BuildScript.cs#L14) and added a test interface. When I run flubu I'm getting this error

```
Flubu v.6.1.1.0
using the build script file path from .flubu file. 'BuildScript/BuildScript.cs'.
File found: D:\researchs\flubucore-versionize-sample\BuildScript\Test\Class1.cs
File found: D:\researchs\flubucore-versionize-sample\BuildScript\Test\Interface1.cs
ScriptError:CS0101: The namespace '<global namespace>' already contains a definition for 'BuildScript'
Unhandled exception. FlubuCore.Scripting.ScriptLoaderExcetpion: Csharp source code file: BuildScript/BuildScript.cs has some compilation errors!
   at FlubuCore.Scripting.ScriptLoader.CompileBuildScript(String buildScriptAssemblyPath, String buildScriptFilePath, IEnumerable`1 references, String code) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 250
   at FlubuCore.Scripting.ScriptLoader.FindAndCreateBuildScriptInstanceAsync(CommandArguments args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 124
   at FlubuCore.Scripting.ScriptProvider.GetBuildScriptAsync(CommandArguments commandArguments, Boolean forceReload) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptProvider.cs:line 26
   at DotNet.Cli.Flubu.Program.FlubuStartup(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 84
   at DotNet.Cli.Flubu.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 36
   at FlubuCore.Tool.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore.Tool\Program.cs:line 9
   at FlubuCore.Tool.Program.<Main>(String[] args)
```

**Comment by sreejith-ms** [comment]
@mzorec I have updated the [flubu sample repo](https://github.com/sreejith-ms/flubucore-versionize-sample/blob/master/BuildScript/BuildScript.cs#L14) and added a test interface. When I run flubu I'm getting this error

```
Flubu v.6.1.1.0
using the build script file path from .flubu file. 'BuildScript/BuildScript.cs'.
File found: D:\flubucore-versionize-sample\BuildScript\Test\Class1.cs
File found: D:\flubucore-versionize-sample\BuildScript\Test\Interface1.cs
ScriptError:CS0101: The namespace '<global namespace>' already contains a definition for 'BuildScript'
Unhandled exception. FlubuCore.Scripting.ScriptLoaderExcetpion: Csharp source code file: BuildScript/BuildScript.cs has some compilation errors!
   at FlubuCore.Scripting.ScriptLoader.CompileBuildScript(String buildScriptAssemblyPath, String buildScriptFilePath, IEnumerable`1 references, String code) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 250
   at FlubuCore.Scripting.ScriptLoader.FindAndCreateBuildScriptInstanceAsync(CommandArguments args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptLoader.cs:line 124
   at FlubuCore.Scripting.ScriptProvider.GetBuildScriptAsync(CommandArguments commandArguments, Boolean forceReload) in c:\jenkins\workspace\FlubuCore\src\FlubuCore\Scripting\ScriptProvider.cs:line 26
   at DotNet.Cli.Flubu.Program.FlubuStartup(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 84
   at DotNet.Cli.Flubu.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\dotnet-flubu\Program.cs:line 36
   at FlubuCore.Tool.Program.Main(String[] args) in c:\jenkins\workspace\FlubuCore\src\FlubuCore.Tool\Program.cs:line 9
   at FlubuCore.Tool.Program.<Main>(String[] args)
```

**Comment by mzorec** [comment]
@sreejith-ms Issue with interfaces is fixed in 6.1.3-preview1. Usages statements inside namespace should work to. It looks like this issue was related with interface issue. I have tested few scenarios and it worked fine.

**Comment by sreejith-ms** [comment]
@mzorec I will check 👍
